src/PIL/ImageFont.py

```diff
@@ -856,7 +857,7 @@ def freetype(font):
     except OSError:
         if not isPath(font):
             raise
-        ttf_filename = os.path.basename(font)
+        ttf_filename = os.fsdecode(os.path.basename(font))
```

Comment:
Similarily, this lookup code also obviously expects `str` paths for everything!

Reply:
Could you add tests for these ImageFont changes?

Reply:
Sorry, but I don't even know what this stuff is for exactly (don't have Windows at all). I just fixed it because it is obviously affected by the same class of bug as pointed out by the previous comment. If that is a problem, I can revert it through.

Reply:
I think this change isn't necessary. I think that `basename` already converts it.
```pycon
>>> from pathlib import Path
>>> import os
>>> filename = os.path.basename(Path("a/b/c"))
>>> filename
'c'
>>> type(filename)
<class 'str'>
```

Reply:
No it is still needed for binary paths (see the added tests):

```py
>>> os.path.basename(b"abcdef/def")
b'def'
>>> os.fsdecode(os.path.basename(b"abcdef/def"))
'def'
```